apis/facebook/api/src/objects/fb_edges.e

```diff
@@ -0,0 +1,155 @@
+note
+	description: "[
+		Object Representing edges - the connections between those "things", such as a Page's Photos, or a Photo's Comments
+		]"
+	date: "$Date$"
+	revision: "$Revision$"
+	EIS: "name=Edges", "src=https://developers.facebook.com/docs/graph-api/overview", "protocol=uri"
+
+class
+	FB_EDGES [G]
+
+create
+	make
+
+feature -- Initialization
+
+	make (a_api: FACEBOOK_JSON)
+		do
+			api := a_api
+			create {ARRAYED_LIST [G]} data.make (0)
+		end
+
+feature -- Access
+
+	data:  LIST [G]
+			-- A list of data nodes.
+
+feature -- Element Change
+
+	set_data (a_data: like data)
+			-- Set `data' with  `a_data'.
+		do
+			data := a_data
+		ensure
+			data_set: data = a_data
+		end
+
+	set_paging (a_paging: like paging)
+			-- Set `paging' with `a_paging'.
+		do
+			paging := a_paging
+		ensure
+			paging_set: paging = a_paging
+		end
+
+	set_summary (a_summary: like summary)
+			-- Set `summary' with `a_summary'.
+		do
+			summary := a_summary
+		ensure
+			summary_set: summary = a_summary
+		end
+
+feature {FB_EDGES} -- Implementation
+
+	paging: detachable FB_PAGING
+			--  pagination.
+
+	summary: detachable FB_SUMMARY
+			-- Aggregated information about the edge, such as counts.
+			-- Specify the fields to fetch in the summary param (like summary=total_count).
+
+	action_back: detachable FUNCTION [detachable FB_EDGES [G]]
+			-- Agent to create a new item of type FB_CONNECTION [G].
+
+	action_forth: detachable FUNCTION [detachable FB_EDGES [G]]
+			-- Agent to create a new item of type FB_CONNECTION [G].
+
+feature -- Change Actions
+
+	set_action_back (a_back: like action_back)
+			-- Set `action_back' with `a_back'.
+		do
+			action_back := a_back
+		ensure
+			action_back_set: action_back = a_back
+		end
+
+	set_action_forth (a_forth: like action_forth)
+			-- Set `action_forth' with `a_forth'.
+		do
+			action_forth := a_forth
+		ensure
+			action_forth_set: action_forth = a_forth
+		end
+
+feature -- Status report
+
+	after: BOOLEAN
```

Comment:
Initially, I design it as an ITERABLE interface, but since it represents pages and every page has a number of entries I decided not to do that. I will add a Note to re-evaluate the current design.